ddtrace/tracer.py

```diff
@@ -121,6 +120,7 @@ def __init__(self, url=None, dogstatsd_url=None):
         self._pid = getpid()
 
         self.enabled = asbool(get_env("trace", "enabled", default=True))
+        self._runtime_metrics_enabled = RuntimeWorker.is_enabled()
```

Comment:
Why storing this?

Reply:
To avoid calling `get_env` inside `start_span`

Reply:
It might be better to cache the lookup directly is `RuntimeWorker.is_enabled`, WDYT?

Reply:
that's an option too

Reply:
I think we can actually do away with the `_runtime_metrics_enabled` attribute. It, along with the `tracer._services` logic, could be moved to `RuntimeWorker` since they're both used exclusively for runtime metrics. 

Then the tag setting and service updating can be done with a `on_start_span` hook using logic that's purely in the `RuntimeWorker`.